[WebGPU Swift] resolveQuerySet workaround is missing from Swift implementation

mwyrzykowski opened the pull request:
#### c9f34b7d16c074ae750d07fc2e337638b580bd9b
<pre>
[WebGPU Swift] resolveQuerySet workaround is missing from Swift implementation
<a href="https://bugs.webkit.org/show_bug.cgi?id=299487">https://bugs.webkit.org/show_bug.cgi?id=299487</a>
<a href="https://rdar.apple.com/161272750">rdar://161272750</a>

Reviewed by Cameron McCormack.

resolveQuerySet support for timestamp queries was in CommandEncoder.mm
but not CommandEncoder.swift

* Source/WebGPU/WebGPU/CommandEncoder.swift:
(WebGPU.counterSampleBuffer):

Canonical link: <a href="https://commits.webkit.org/301006@main">https://commits.webkit.org/301006@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a2a4f0ec66380e40a3adb169fa685fb35d3b1564

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124584 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44256 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/34990 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131422 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76531 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/5d4566aa-6eee-4e2d-b0ea-7e074a4bd6ab) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/44960 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/52827 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/94773 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/62850 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/2b9c67b3-96f8-4a28-bfa9-2a978143934d) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127538 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/35845 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111413 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75415 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/3b564f92-4ede-4be0-848e-8c8daf746d83) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/34783 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/74901 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105600 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/29799 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134091 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51436 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39255 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103254 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/51844 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107635 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103033 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26242 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48394 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26657 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48370 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51307 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57103 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/50710 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54065 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52396 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

heycam commented on Source/WebGPU/WebGPU/CommandEncoder.swift:
```diff
@@ -1847,15 +1847,35 @@ extension WebGPU.CommandEncoder {
                 return
             }
             blitCommandEncoder.copy(from: querySet.visibilityBuffer(), sourceOffset: sourceOffset, to: destination.buffer(), destinationOffset: destinationOffsetChecked, size: size)
         }
+        if querySet.type().rawValue == WGPUQueryType_Timestamp.rawValue {
+            self.finalizeBlitCommandEncoder()
```
Should we duplicate the comment about reverting the timestamp counter workaround here?

mwyrzykowski replied:
yes, will add that

heycam commented on Source/WebGPU/WebGPU/CommandEncoder.swift:
```diff
@@ -1847,15 +1847,35 @@ extension WebGPU.CommandEncoder {
+            var counterSampleBuffer: MTLCounterSampleBuffer? = nil
+            counterSampleBuffer = querySet.counterSampleBufferWithOffset().first
+            let counterSampleBufferOffset = querySet.counterSampleBufferWithOffset().second
+            unsafe m_blitCommandEncoder.resolveCounters(
+                counterSampleBuffer!,
+                range: Range(uncheckedBounds: (Int(firstQuery + counterSampleBufferOffset), Int(queryCount))),
+                destinationBuffer: destination.buffer(),
+                destinationOffset: Int(destinationOffset)
+            )
```
Since we weren't calling this at all before, maybe the commit message title should be "add support for timestamp counters" or so?

mwyrzykowski committed:
[WebGPU Swift] resolveQuerySet workaround is missing from Swift implementation
https://bugs.webkit.org/show_bug.cgi?id=299487
rdar://161272750

Reviewed by Cameron McCormack.

resolveQuerySet support for timestamp queries was in CommandEncoder.mm
but not CommandEncoder.swift

* Source/WebGPU/WebGPU/CommandEncoder.swift:
(WebGPU.counterSampleBuffer):

Canonical link: https://commits.webkit.org/301006@main
